From 1baff522c76b8b7b34f30d9601ceeb70266590ed Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:27 -0400
Subject: [tools] perf/intlist: Add priv member

Message-id: <1408462094-14194-159-git-send-email-jolsa@redhat.com>
Patchwork-id: 88122
O-Subject: [PATCH RHEL7.1 BZ1131394 158/325] perf intlist: Add priv member
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 2969b12993ca7a8b9692048431e075a67815002d
Author: David Ahern <dsahern@gmail.com>
Date: Sat Sep 28 13:13:02 2013 -0600

description
===========
Allows commands to leverage intlist infrastructure for opaque
structures.

For example an upcoming perf-trace change will use this as a means of
tracking syscalls statistics by task.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/intlist.c b/tools/perf/util/intlist.c
index 11a8d86..826d7b3 100644
--- a/tools/perf/util/intlist.c
+++ b/tools/perf/util/intlist.c
@@ -20,6 +20,7 @@ static struct rb_node *intlist__node_new(struct rblist *rblist __maybe_unused,
 
  if (node != NULL) {
   node->i = i;
+  node->priv = NULL;
   rc = &node->rb_node;
  }
 
diff --git a/tools/perf/util/intlist.h b/tools/perf/util/intlist.h
index 62351da..0eb00ac 100644
--- a/tools/perf/util/intlist.h
+++ b/tools/perf/util/intlist.h
@@ -9,6 +9,7 @@
 struct int_node {
  struct rb_node rb_node;
  int i;
+ void *priv;
 };
 
 struct intlist {
-- 
1.7.1